crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -0,0 +1,194 @@
+  state = bssl::UniquePtr<fips_service_indicator_state>(awslc_fips_service_indicator_get_state());
+
+  ASSERT_TRUE(state.get());
+  ASSERT_EQ(state.get()->counter,0);
+  ASSERT_EQ(state.get()->serviceID, fips_approved_no_state);
+
+  int counter = awslc_fips_service_indicator_get_counter();
+  ASSERT_EQ(counter, state.get()->counter);
+
+  // Call a random approved service.
+  EVP_AEAD_CTX aead_ctx;
+  EVP_AEAD_CTX_zero(&aead_ctx);
+  uint8_t output[256];
+  size_t out_len;
+  ASSERT_TRUE(EVP_AEAD_CTX_init(&aead_ctx, EVP_aead_aes_128_gcm_randnonce(), kAESKey,
+                         sizeof(kAESKey), 0, nullptr));
+  ASSERT_TRUE(EVP_AEAD_CTX_seal(&aead_ctx, output, &out_len, sizeof(output), nullptr,
+             0, kPlaintext, sizeof(kPlaintext), nullptr, 0));
+
+  ASSERT_TRUE(awslc_fips_check_service_approved(counter, fips_approved_evp_aes_128_gcm));
+
+  // Check state and counter after using an approved service.
+  state = bssl::UniquePtr<fips_service_indicator_state>(awslc_fips_service_indicator_get_state());
+  ASSERT_TRUE(state.get());
+  ASSERT_EQ(state.get()->counter,1);
+  ASSERT_EQ(state.get()->serviceID, fips_approved_evp_aes_128_gcm);
+
+  counter = awslc_fips_service_indicator_get_counter();
+  ASSERT_EQ(counter, state.get()->counter);
+}
+
+TEST(ServiceIndicatorTest, AESCBC) {
+  int counter = awslc_fips_service_indicator_get_counter();
+
+  AES_KEY aes_key;
+  uint8_t aes_iv[16];
+  uint8_t output[256];
+
+  // AES-CBC Encryption KAT
+  memcpy(aes_iv, kAESIV, sizeof(kAESIV));
+  if (AES_set_encrypt_key(kAESKey, 8 * sizeof(kAESKey), &aes_key) != 0) {
+    fprintf(stderr, "AES_set_encrypt_key failed.\n");
+    return;
+  }
+  AES_cbc_encrypt(kPlaintext, output, sizeof(kPlaintext), &aes_key, aes_iv,
+                  AES_ENCRYPT);
+  if (!check_test(kAESCBCCiphertext, output, sizeof(kAESCBCCiphertext),
+                  "AES-CBC Encryption KAT")) {
+    return;
+  }
+
+  ASSERT_TRUE(awslc_fips_check_service_approved(counter, fips_approved_evp_aes_128_cbc));
+  counter = awslc_fips_service_indicator_get_counter();
+
+  // AES-CBC Decryption KAT
+  memcpy(aes_iv, kAESIV, sizeof(kAESIV));
+  if (AES_set_decrypt_key(kAESKey, 8 * sizeof(kAESKey), &aes_key) != 0) {
+    fprintf(stderr, "AES_set_decrypt_key failed.\n");
+    return;
+  }
+  AES_cbc_encrypt(kAESCBCCiphertext, output, sizeof(kAESCBCCiphertext),
+                  &aes_key, aes_iv, AES_DECRYPT);
+  if (!check_test(kPlaintext, output, sizeof(kPlaintext),
+                  "AES-CBC Decryption KAT")) {
+    return;
+  }
+
+  ASSERT_TRUE(awslc_fips_check_service_approved(counter, fips_approved_evp_aes_128_cbc));
+}
+
+TEST(ServiceIndicatorTest, AESGCM) {
+  bssl::ScopedEVP_AEAD_CTX aead_ctx;
+
+  uint8_t output[256];
+  int counter = awslc_fips_service_indicator_get_counter();
+
+  size_t out_len;
+  ASSERT_TRUE(EVP_AEAD_CTX_init(aead_ctx.get(), EVP_aead_aes_128_gcm_randnonce(), kAESKey,
+                         sizeof(kAESKey), 0, nullptr));
+
+  // AES-GCM Encryption KAT
```

Comment:
This is not a KAT (known-answer test), but can be consistency test if we use open to check that the plaintext is the same as the ciphertext and that it's approved as well (counter increased again).

Reply:
Added a decryption test right after